### PR TITLE
Ecdc 1284 check units

### PR DESCRIPTION
**Pull request by SkyToGround**
Check and warn about changing units.

I have added no unit or system tests for this new functionality as I think it is too much work for too little effort to test this. I can add one though if you insist.

### CODE REVIEW

File: forwarder/update_handlers/ca_update_handler.py [hunk 1]
```diff
@@ -32,6 +34,20 @@ def __init__(
...
 
+        ctrl_sub = self._pv.subscribe(data_type="control")
+        ctrl_sub.add_callback(self._unit_callback)
+
+    def _unit_callback(self, sub, response: ReadNotifyResponse):
+        old_unit = self._current_unit
+        try:
+            self._current_unit = response.metadata.units.decode("utf-8")
+        except AttributeError:
+            return
```
Comment (kmurica) [inline forwarder/update_handlers/ca_update_handler.py:45]: Would it be useful to add a self._logger.error("some message about attribute error") here?

File: forwarder/update_handlers/pva_update_handler.py [hunk 2]
```diff
@@ -31,7 +31,16 @@ def __init__(
             notify_disconnect=True,
         )
 
-    def _monitor_callback(self, response: Union[Value, Exception]):
+    def _monitor_callback(self, response):
+        old_unit = self._unit
+        try:
+            self._unit = response.display.units
+        except AttributeError:
+            pass
```
Comment (kmurica) [inline forwarder/update_handlers/pva_update_handler.py:39]: Same as above.

**Comment by kmurica** [comment]
Minor comment. Looks good otherwise. Merge if you want or add some logging to attribute error and merge. I will approve it either way.

**Review by kmurica — APPROVED** [review]
See comment. Otherwise looks good.

File: forwarder/update_handlers/ca_update_handler.py (hunk 1 above)
Comment (SkyToGround) [reply forwarder/update_handlers/ca_update_handler.py:45]: I do not think so. Checking actual PVs in use, there are A LOT (maybe even a majority) that does not have a unit set which will then result in an AttributeError. This change will thus result in a message that will be too common and thus useless.

File: forwarder/update_handlers/pva_update_handler.py (hunk 2 above)
Comment (SkyToGround) [reply forwarder/update_handlers/pva_update_handler.py:39]: See above. :-)

**Pull request merged**